File: shared/src/main/java/com/alta_v2/SharedInjector.java
package com.alta_v2;

import com.alta_v2.aop.executionTime.PrintExecutionTime;
import com.alta_v2.aop.executionTime.PrintExecutionTimeHandler;
import com.google.inject.AbstractModule;
import com.google.inject.matcher.Matchers;

public class SharedInjector extends AbstractModule {

    @Override
    protected void configure() {
        bindInterceptor(Matchers.any(), Matchers.annotatedWith(PrintExecutionTime.class), new PrintExecutionTimeHandler());
    }

}

File: shared/src/main/java/com/alta_v2/SharedInjector.kt
package com.alta_v2

import com.alta_v2.aop.executionTime.PrintExecutionTime
import com.alta_v2.aop.executionTime.PrintExecutionTimeHandler
import com.google.inject.AbstractModule
import com.google.inject.matcher.Matchers

class SharedInjector : AbstractModule() {
    override fun configure() {
        bindInterceptor(Matchers.any(), Matchers.annotatedWith(PrintExecutionTime::class.java), PrintExecutionTimeHandler())
    }
}